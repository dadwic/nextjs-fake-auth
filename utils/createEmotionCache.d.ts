import type { EmotionCache } from "@emotion/utils";

export const isBrowser: boolean;

export default function createEmotionCache(): EmotionCache;
