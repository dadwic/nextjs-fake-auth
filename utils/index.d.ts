import type { Theme } from "@mui/material";

export const theme: Theme;

export const isServer: boolean;

export function fetcher(url: string): Promise<Response>;

export function a11yProps(index: number): object;
