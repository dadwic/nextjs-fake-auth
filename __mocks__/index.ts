import { LoginFormInput, SignUpFormInput } from "interfaces";

export const loginCredential: LoginFormInput = {
  email: "email@example.com",
  password: "123456",
};

export const userData: SignUpFormInput = {
  firstName: "John",
  lastName: "Doe",
  email: "email@example.com",
  password: "123456",
};
