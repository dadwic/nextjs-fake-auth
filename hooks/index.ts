import { useEffect, useState } from "react";

export function useStickyState(defaultValue: string, key: string) {
  const [value, setValue] = useState(() => {
    const stickyValue = window.localStorage.getItem(key);
    return stickyValue !== null ? stickyValue : defaultValue;
  });
  useEffect(() => {
    window.localStorage.setItem(key, value);
  }, [key, value]);
  return [value, setValue];
}
