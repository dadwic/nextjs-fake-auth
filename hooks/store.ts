import { useMemo } from "react";
import { createStore, applyMiddleware, Action } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

let store: any;

const exampleInitialState = {
  email: null,
};

export const actionTypes = {
  LOGIN: "LOGIN",
  LOGOUT: "LOGOUT",
};

// REDUCERS
export const reducer = (state = exampleInitialState, action: Action) => {
  switch (action.type) {
    case actionTypes.LOGIN:
      return {
        ...state,
        email: state.email,
      };
    case actionTypes.LOGOUT:
      return {
        ...state,
        count: null,
      };
    case actionTypes.LOGOUT:
      return {
        ...state,
        count: exampleInitialState.email,
      };
    default:
      return state;
  }
};

// ACTIONS
export const logout = () => {
  return { type: actionTypes.LOGOUT };
};

export const login = (data: { email: string }) => {
  return { type: actionTypes.LOGIN, data };
};

const persistConfig = {
  key: "primary",
  storage,
  whitelist: ["exampleData"], // place to select which state you want to persist
};

const persistedReducer = persistReducer(persistConfig, reducer);

function makeStore(initialState = exampleInitialState) {
  return createStore(
    persistedReducer,
    initialState as any,
    composeWithDevTools(applyMiddleware())
  );
}

export const initializeStore = (preloadedState: any) => {
  let _store = store ?? makeStore(preloadedState);

  // After navigating to a page with an initial Redux state, merge that state
  // with the current state in the store, and create a new store
  if (preloadedState && store) {
    _store = makeStore({
      ...store.getState(),
      ...preloadedState,
    });
    // Reset the current store
    store = undefined;
  }

  // For SSG and SSR always create a new store
  if (typeof window === "undefined") return _store;
  // Create the store once in the client
  if (!store) store = _store;

  return _store;
};

export function useStore(initialState: any) {
  const store = useMemo(() => initializeStore(initialState), [initialState]);
  return store;
}
