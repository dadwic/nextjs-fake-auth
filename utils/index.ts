import { createTheme } from "@mui/material/styles";
import { red } from "@mui/material/colors";

// Create a theme instance.
export const theme = createTheme({
  components: {
    MuiLink: {
      styleOverrides: {
        root: { color: "common.black", textDecoration: "none" },
      },
    },
    MuiTab: {
      styleOverrides: {
        root: {
          textTransform: "none",
        },
      },
    },
    MuiButton: {
      styleOverrides: {
        fullWidth: {
          margin: "24px 0",
          borderRadius: 24,
          height: 48,
        },
      },
    },
    MuiInputLabel: {
      styleOverrides: {
        root: {
          width: "100%",
          textAlign: "center",
        },
        shrink: {
          textAlign: "left",
        },
      },
    },
  },
  palette: {
    primary: {
      main: "#333131",
    },
    secondary: {
      main: "#3D74B8",
    },
    error: {
      main: red.A400,
    },
  },
});

export const isServer = typeof window === "undefined";

// SWR fetcher
export const fetcher = (url: string) => fetch(url).then((res) => res.json());

export function a11yProps(index: number) {
  return {
    id: `simple-tab-${index}`,
    "aria-controls": `simple-tabpanel-${index}`,
  };
}
