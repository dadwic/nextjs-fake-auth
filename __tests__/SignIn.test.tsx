import { loginCredential, userData } from "../__mocks__";
import { renderWithProviders, mockNextUseRouter } from "utils/test-utils";
import { screen, fireEvent } from "@testing-library/react";
import AuthPage from "pages/auth/[slug]";
import { login } from "hooks/store";

describe("SignIn Page", () => {
  mockNextUseRouter({
    route: "/",
    asPath: "/auth/login",
    pathname: "/auth/[slug]",
    query: { slug: "login" },
  });

  test("login after clicking the login button", async () => {
    const { store, container } = renderWithProviders(<AuthPage />);
    expect(container).toMatchSnapshot();

    fireEvent.input(screen.getByLabelText("E-mail Adresin *"), {
      target: {
        value: loginCredential.email,
      },
    });
    fireEvent.input(screen.getByLabelText("Şifren *"), {
      target: {
        value: loginCredential.password,
      },
    });
    // Submit the form
    fireEvent.submit(screen.getByTestId("submit"));

    // Dispatch login action
    store.dispatch(login(loginCredential.email));

    // Checking the login successful alert
    // expect(await screen.findAllByRole("alert")).toHaveLength(1);

    // Get store states
    const { email } = store.getState();
    expect(email).toBe(loginCredential.email);
  });
});

describe("SignUp Page", () => {
  mockNextUseRouter({
    route: "/",
    asPath: "/auth/signup",
    pathname: "/auth/[slug]",
    query: { slug: "signup" },
  });

  test("create a user and login after clicking the signup button", async () => {
    const { store, container } = renderWithProviders(<AuthPage />);
    expect(container).toMatchSnapshot();

    fireEvent.input(screen.getByLabelText("Adın *"), {
      target: {
        value: userData.firstName,
      },
    });
    fireEvent.input(screen.getByLabelText("Soyadın *"), {
      target: {
        value: userData.lastName,
      },
    });
    fireEvent.input(screen.getByLabelText("E-mail Adresin *"), {
      target: {
        value: loginCredential.email,
      },
    });
    fireEvent.input(screen.getByLabelText("Şifren *"), {
      target: {
        value: loginCredential.password,
      },
    });
    // Submit the form
    fireEvent.submit(screen.getByTestId("submit"));

    // Dispatch login action
    store.dispatch(login(loginCredential.email));

    // Get store states
    const { email } = store.getState();
    expect(email).toBe(loginCredential.email);
  });
});
