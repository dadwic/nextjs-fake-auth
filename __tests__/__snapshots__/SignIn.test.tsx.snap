// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Page login after clicking the login button 1`] = `
<div>
  <main
    class="MuiGrid-root MuiGrid-container css-1jrn2fe-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-13eeoot-MuiGrid-root"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-1y8ugea"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1hmptw2-MuiTypography-root"
          >
            Merhaba
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiTypography-gutterBottom css-ujp2qg-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet consectetur adipiscing elit.
          </p>
          <div
            class="MuiBox-root css-8atqhb"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="basic tabs example"
                  class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="simple-tabpanel-0"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1g4fz9n-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-0"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Giriş Yap
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-1"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1g4fz9n-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-1"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    Üye Ol
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-fuzn5c-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-labelledby="simple-tab-0"
            hidden=""
            id="simple-tabpanel-0"
            role="tabpanel"
          />
          <div
            aria-labelledby="simple-tab-1"
            id="simple-tabpanel-1"
            role="tabpanel"
          >
            <div
              class="MuiBox-root css-1sf3xto"
            >
              <form
                class="MuiBox-root css-164r41r"
                novalidate=""
              >
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="firstName"
                    id="firstName-label"
                  >
                    Adın
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-4eefpt-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="fname"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="firstName"
                      name="firstName"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          Adın
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="lastName"
                    id="lastName-label"
                  >
                    Soyadın
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-4eefpt-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="lname"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="lastName"
                      name="lastName"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          Soyadın
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="email"
                    id="email-label"
                  >
                    E-mail Adresin
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-4eefpt-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="email"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="email"
                      name="email"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          E-mail Adresin
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="password"
                    id="password-label"
                  >
                    Şifren
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1gmwxzp-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      id="password"
                      name="password"
                      required=""
                      type="password"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="toggle password visibility"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="VisibilityIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          Şifren
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-17fzijo-MuiTypography-root"
                >
                  Şifrenizde en az bir harf, sayı veya özel karakter var Içermelidir. Ayrıca şifreniz en az 8 karakter olmalıdır. meydana gelmelidir.
                </h6>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-66puty-MuiButtonBase-root-MuiButton-root"
                  data-testid="submit"
                  tabindex="0"
                  type="submit"
                >
                  Üye Ol
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </form>
            </div>
          </div>
          <footer
            class="MuiBox-root css-y1gt6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Made with ❤️ by
               
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-gsdc4g-MuiTypography-root-MuiLink-root"
                href="https://github.com/dadwic"
                rel="noopener noreferrer"
                target="_blank"
              >
                @dadwic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Copyright © 2022
            </p>
          </footer>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`SignUp Page create a user and login after clicking the signup button 1`] = `
<div>
  <main
    class="MuiGrid-root MuiGrid-container css-1jrn2fe-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-13eeoot-MuiGrid-root"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-1y8ugea"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1hmptw2-MuiTypography-root"
          >
            Merhaba
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiTypography-gutterBottom css-ujp2qg-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet consectetur adipiscing elit.
          </p>
          <div
            class="MuiBox-root css-8atqhb"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="basic tabs example"
                  class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="simple-tabpanel-0"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1g4fz9n-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-0"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Giriş Yap
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-1"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1g4fz9n-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-1"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    Üye Ol
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-fuzn5c-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-labelledby="simple-tab-0"
            hidden=""
            id="simple-tabpanel-0"
            role="tabpanel"
          />
          <div
            aria-labelledby="simple-tab-1"
            id="simple-tabpanel-1"
            role="tabpanel"
          >
            <div
              class="MuiBox-root css-1sf3xto"
            >
              <form
                class="MuiBox-root css-164r41r"
                novalidate=""
              >
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="firstName"
                    id="firstName-label"
                  >
                    Adın
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-4eefpt-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="fname"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="firstName"
                      name="firstName"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          Adın
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="lastName"
                    id="lastName-label"
                  >
                    Soyadın
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-4eefpt-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="lname"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="lastName"
                      name="lastName"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          Soyadın
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="email"
                    id="email-label"
                  >
                    E-mail Adresin
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-4eefpt-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="email"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="email"
                      name="email"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          E-mail Adresin
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-y3gpis-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="password"
                    id="password-label"
                  >
                    Şifren
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-dqj5rd-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1gmwxzp-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      id="password"
                      name="password"
                      required=""
                      type="password"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="toggle password visibility"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="VisibilityIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          Şifren
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-17fzijo-MuiTypography-root"
                >
                  Şifrenizde en az bir harf, sayı veya özel karakter var Içermelidir. Ayrıca şifreniz en az 8 karakter olmalıdır. meydana gelmelidir.
                </h6>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-66puty-MuiButtonBase-root-MuiButton-root"
                  data-testid="submit"
                  tabindex="0"
                  type="submit"
                >
                  Üye Ol
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </form>
            </div>
          </div>
          <footer
            class="MuiBox-root css-y1gt6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Made with ❤️ by
               
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-gsdc4g-MuiTypography-root-MuiLink-root"
                href="https://github.com/dadwic"
                rel="noopener noreferrer"
                target="_blank"
              >
                @dadwic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Copyright © 2022
            </p>
          </footer>
        </div>
      </div>
    </div>
  </main>
</div>
`;
